### PR TITLE
Vectorization and other changes to value_at

### PR DESCRIPTION
**Pull request by dana89co**
1. Vectorization of value_at_lq only (for value_at_lb it was not necessary)
2. Add error if any value in vector is negative, NA or Inf 
3. Tests for new value_at_lq and new check functions

### CODE REVIEW

**Comment by dana89co** [comment]
Task linked: [CU-8686rcmv0 vectorize inefficient functions in wbpip](https://app.clickup.com/t/8686rcmv0)

File: R/gd_compute_pip_stats_lq.R [hunk 1]
```diff
@@ -303,6 +303,34 @@ gd_compute_gini_lq <- function(A, B, C, e, m, n, r) {
...
+
+  # Check for NA, Inf and negative values in x
+  check_NA_Inf_values(x)
+  check_neg_values(x)
+
+  # Calculations
+  e <- -(A + B + C + 1)
+  m <- (B^2) - (4 * A)
+  n <- (2 * B * e) - (4 * C)
+  temp <- pmax((m * x^2) + (n * x) + (e^2),0)
```
Comment (randrescastaneda) [inline R/gd_compute_pip_stats_lq.R]: Hi @dana89co , 

I checked the speed of `pmax()` and found it slower than regular indexing (see below). Try to use indexing as often as possible for vectorizations. It is really fast. 

```r

x <- seq(0, 1, 1e-3)

A <- 0.6
B <- 0.3
C <- 0.4


e <- -(A + B + C + 1)
m <- (B^2) - (4 * A)
n <- (2 * B * e) - (4 * C)  
  


bench <- microbenchmark::microbenchmark(
  times = 100,
  index = {
    temp2 <- (m * x^2) + (n * x) + (e^2)
    temp2[temp2 < 0] <- 0  
    },
  pmax = {
    temp1 <- pmax((m * x^2) + (n * x) + (e^2),0) 
    }
)
if (requireNamespace("highcharter")) {
  hc_dt <- highcharter::data_to_boxplot(bench,
                                        time,
                                        expr,
                                        add_outliers = FALSE,
                                        name = "Time in milliseconds")

  highcharter::highchart() |>
  highcharter::hc_xAxis(type = "category") |>
  highcharter::hc_chart(inverted=TRUE) |>
  highcharter::hc_add_series_list(hc_dt)
  
} else {
  boxplot(bench, outline = FALSE)
}
```

image

Comment (randrescastaneda) [reply R/gd_compute_pip_stats_lq.R]: Just make the change and push again. You don't need to create another PR or check out a new branch. Thanks.

Comment (dana89co) [reply R/gd_compute_pip_stats_lq.R]: Makes sense 👍  Thanks for checking this inefficiency. I will change it and take that into account from now on.

**Comment by randrescastaneda** [comment]
Hola @dana89co , 

Could you please change this PR to merge into `dev_stage2` branch? Also, please make sure the PR is free of conflicts. Thank you!

**Comment by dana89co** [comment]
Hola @randrescastaneda, I should have closed this PR before because, if I am not mistaken, these changes were already included on dev_stage2 when I did the following PR: https://github.com/PIP-Technical-Team/wbpip/pull/244. I am sorry.

**Pull request closed without merging**